feat: use exported targets

wep21 opened the pull request:
- revert the changes for ament cmake auto and use exported targets instead.

Commit:
Revert "Cleanup CMakeLists.txt with ament_cmake_auto (#57)"

This reverts commit bea4f9b90e914446d82ea6b2710d03efd37c0e66.

Commit:
feat: use exported targets

Signed-off-by: Daisuke Nishimatsu <border_goldenmarket@yahoo.co.jp>

wep21 commented:
@clalancette @hidmic Could anyone review this PR?

clalancette commented on CMakeLists.txt:
```diff
@@ -1,33 +1,94 @@
+  DESTINATION lib/${PROJECT_NAME}
+)
+
+install(DIRECTORY include
+  DESTINATION include
+)
```
We should change this to a different directory so that overrides work properly.  Please make this:

```suggestion
install(DIRECTORY include
  DESTINATION include/${PROJECT_NAME}
)
```

clalancette commented on CMakeLists.txt:
```diff
@@ -1,33 +1,94 @@
-  INSTALL_TO_SHARE
-    launch
-)
+ament_package()
```
We are missing the export of the dependencies.  This should be something like:

```
ament_export_dependencies(laser_geometry message_filters rclcpp sensor_msgs tf2_ros)
```

wep21 replied:
@clalancette I added missing ament_export_dependencies in https://github.com/ros-perception/pointcloud_to_laserscan/pull/69/commits/be9226b82d38274a2ae02c841826a60ff48e5551.

clalancette commented on CMakeLists.txt:
```diff
@@ -1,33 +1,94 @@
+install(TARGETS
+  laserscan_to_pointcloud
+  pointcloud_to_laserscan
+  RUNTIME DESTINATION lib
```
```suggestion
  RUNTIME DESTINATION bin
```

(even though we don't use this, that is consistent with what we do elsewhere)

clalancette commented on CMakeLists.txt:
```diff
@@ -1,33 +1,94 @@
+  "${sensor_msgs_TARGETS}"
+)
+
+install(TARGETS
+  laserscan_to_pointcloud
+  pointcloud_to_laserscan
```
We should probably EXPORT this so that downstream projects could link against it if they wanted.  So this should be:

```suggestion
install(TARGETS
  laserscan_to_pointcloud
  pointcloud_to_laserscan
  EXPORT export_${PROJECT_NAME}
```

clalancette reviewed (changes requested):
I think this is the right direction to go.

I've left a few fairly minor things to fixed.  Once those are good, I'm happy to approve.

wep21 committed:
fix include install directory

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

wep21 committed:
export targets

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

wep21 committed:
fix runtime destination

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

Commit:
add missing ament export dependencies

Signed-off-by: Daisuke Nishimatsu <border_goldenmarket@yahoo.co.jp>

clalancette commented on CMakeLists.txt:
```diff
@@ -1,33 +1,106 @@
+  rclcpp_components::component
+  tf2::tf2
+  tf2_ros::tf2_ros
+  # use exported target after https://github.com/ros2/geometry2/pull/536 is merged
+  # tf2_sensor_msgs::tf2_sensor_msgs
```
So when I tried to build this, it failed with:

```
pc_to_laserscan_ws/src/pointcloud_to_laserscan/src/pointcloud_to_laserscan_node.cpp:52:10: fatal error: tf2_sensor_msgs/tf2_sensor_msgs.h: No such file or directory
   52 | #include "tf2_sensor_msgs/tf2_sensor_msgs.h"
```

A workaround would be to add `${tf2_sensor_msgs_INCLUDE_DIRS}` to the `target_include_directories`.  We can either do that, or we can wait for https://github.com/ros2/geometry2/pull/536 to be merged.  I leave the decision to you, either is fine with me.

wep21 replied:
@clalancette I decided to add workaround until https://github.com/ros2/geometry2/pull/536 is merged. https://github.com/ros-perception/pointcloud_to_laserscan/pull/69/commits/f4e87810020db03a9687966897c979440fcf8b42
Thank you for letting me know.

wep21 replied:
@clalancette friendly ping

Commit:
add tf2 sensor msgs include dir for workaround

Signed-off-by: Daisuke Nishimatsu <border_goldenmarket@yahoo.co.jp>

wep21 commented:
@clalancette friendly ping

wep21 commented:
@clalancette friendly ping

wep21 commented:
@clalancette friendly ping

clalancette committed:
Fixes for the latest.

Now that we merged in the geometry2 PR, we can use the
target as appropriate.  Also fix a warning when building
against rolling.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette reviewed (approved):
@wep21 I ended up going back to the `tf2_sensor_msg::tf2_sensor_msgs` target, as we've merged in the geometry2 PR in the meantime.  I also fixes a warning when building.  This now looks good to me, thank you!

clalancette commented:
@ros-pull-request-builder retest this please
